api/hasura/actions/create_user.ts

```diff
@@ -0,0 +1,100 @@
+import type { VercelRequest, VercelResponse } from '@vercel/node';
+
+import { gql } from '../../../api-lib/Gql';
+import { verifyHasuraAdminMiddleware } from '../../../api-lib/validate';
+import { profiles_constraint } from '../../../src/lib/gql/zeusHasuraAdmin';
+import { createUserSchemaInput } from '../../../src/lib/zod';
+
+async function handler(request: VercelRequest, response: VercelResponse) {
+  // Input Validation
+  const result = await createUserSchemaInput.safeParseAsync(request.body.input);
+  // a `switch` gets typescript to behave soundly
```

Comment:
"to behave soundly"... what's going on here? Is the `if(result.success) == true` not working?

Reply:
@crabsinger the code you commented is good js, but fails the typescript typechecker.


NBD because @exrhizo proposed some syntax that works as expected above.